lib/note-editor/index.tsx

```diff
@@ -161,13 +162,15 @@ export class NoteEditor extends Component {
   }
 }
 
-const mapStateToProps = ({ appState: state, settings }) => ({
+const mapStateToProps = ({ appState: state, settings, ui: { note } }) => ({
   allTags: state.tags,
   filter: state.filter,
   fontSize: settings.fontSize,
   editorMode: state.editorMode,
   isEditorActive: !state.showNavigation,
+  note,
   revision: state.revision,
+  tags: get(note, 'data.tags', []),
```

Comment:
_note_ that with our TypeScript annotations we can relax our runtime safety here - we can call `note.data.tags` directly. If we try and call that by accident on `Note` instead of `NoteEntity` it will alert us.

if `tags` is possibly missing then we'll need to update our type or make sure to add it on entry into the app

Reply:
Noice, changed

Reply:
if it was changed it doesn't appear to have made it here

as a benefit of removing `get()` - since this is `mapStateToProps` we will trigger re-renders as long as we return `[]` instead of some array that is reference-equal to the one before it (such as `note.data.tags` when the tags don't change).

Reply:
I see now too that `note` may be `null`, though we can employ the optional-chaining trick from below. I see why you did this now.

```ts
tags: note?.data.tags ?? []
```